sys/linux/socket_ip_tunnel.txt

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/linux/if_tunnel.h>
+include <uapi/linux/ip6_tunnel.h>
+include <uapi/linux/if.h>
+include <uapi/linux/in.h>
+include <uapi/linux/ip.h>
+include <net/ip.h>
+
+# Fallback tunnels and placeholder names "syztnl*".
+ipv4_tunnel_names = "ip_vti0", "tunl0", "gre0", "gretap0", "erspan0", "syztnl0", "syztnl1", "syztnl2"
+ipv6_tunnel_names = "ip6tnl0", "ip6_vti0", "ip6gre0", "sit0", "syztnl0", "syztnl1", "syztnl2"
+
+ip_tunnel_protocols = 0, IPPROTO_IPIP, IPPROTO_GRE, IPPROTO_IPV6
+ip_tunnel_io_flags = GRE_CSUM, GRE_ROUTING, GRE_KEY, GRE_SEQ, GRE_STRICT, GRE_REC, GRE_ACK, GRE_FLAGS, GRE_VERSION, VTI_ISVTI
+
+type ipv4_header_only ipv4_packet_t[flags[ip_tunnel_protocols, int8], void]
+
+# IPv4 tunnel parameters
+ip_tunnel_parm {
+	name	string[ipv4_tunnel_names, IFNAMSIZ]
+	link	int32
+	i_flags	flags[ip_tunnel_io_flags, int16be]
+	o_flags	flags[ip_tunnel_io_flags, int16be]
+	i_key	int32be
+	o_key	int32be
+	iph	ipv4_header_only
+}
+
+ipv6_tunnel_flags = IP6_TNL_F_IGN_ENCAP_LIMIT, IP6_TNL_F_USE_ORIG_TCLASS, IP6_TNL_F_USE_ORIG_FLOWLABEL, IP6_TNL_F_MIP6_DEV, IP6_TNL_F_RCV_DSCP_COPY, IP6_TNL_F_USE_ORIG_FWMARK, IP6_TNL_F_ALLOW_LOCAL_REMOTE
+
+# IPv6 tunnel parameters
+ip6_tnl_parm2 {
+	name		string[ipv6_tunnel_names, IFNAMSIZ]
+	link		int32
+	proto		int8
```

Comment:
Should this refer to some flags?

Reply:
It should be `ip_tunnel_protocols`. Thanks for the catch!